server/util.go

```diff
@@ -55,11 +57,24 @@ func NewContext(v *viper.Viper, config *tmcfg.Config, logger log.Logger) *Contex
 // the application configuration. Command handlers can fetch the server Context
 // to get the Tendermint configuration or to get access to Viper.
 func InterceptConfigsPreRunHandler(cmd *cobra.Command) error {
-	rootViper := viper.New()
+	serverCtx := NewDefaultContext()
+
+	rootViper := serverCtx.Viper
 	rootViper.BindPFlags(cmd.Flags())
 	rootViper.BindPFlags(cmd.PersistentFlags())
 
-	serverCtx := NewDefaultContext()
+	// Get the executable name and configure the viper instance so that environmental
+	// variables are checked based off that name. The underscore character is used
+	// as a separator
+	executableName, err := os.Executable()
+	if err != nil {
+		return err
+	}
+	basename := path.Base(executableName)
+	rootViper.SetEnvPrefix(basename)
+	rootViper.SetEnvKeyReplacer(strings.NewReplacer(".", "_", "-", "_"))
+	rootViper.AutomaticEnv()
+
 	config, err := interceptConfigs(serverCtx, rootViper)
```

Comment:
I see no reason why `interceptConfigs` can't just take a `serverCtx` since the `serverCtx` already has the root viper configuration object.

Reply:
After looking at this a while I'm thinking it is the opposite. There is no reason to pass in the context, the `interceptConfigs` function only really needs the Viper instance.

Reply:
Sure!